src/Microsoft.AspNetCore.OData/Batch/ODataBatchContent.cs

```diff
@@ -28,9 +32,29 @@ public partial class ODataBatchContent
         /// <param name="requestContainer">The dependency injection container for the request.</param>
         /// <remarks>This signature uses types that are AspNetCore-specific.</remarks>
         public ODataBatchContent(IEnumerable<ODataBatchResponseItem> responses, IServiceProvider requestContainer)
+            : this(responses, requestContainer,  null/*contentType*/)
+        {
+        }
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataBatchContent"/> class.
+        /// </summary>
+        /// <param name="responses">The batch responses.</param>
+        /// <param name="requestContainer">The dependency injection container for the request.</param>
+        /// <param name="contentType">The response content type.</param>
+        /// <remarks>This signature uses types that are AspNetCore-specific.</remarks>
+        public ODataBatchContent(IEnumerable<ODataBatchResponseItem> responses, IServiceProvider requestContainer,
+           string contentType)
         {
             Initialize(responses, requestContainer);
-            Headers[HeaderNames.ContentType] = String.Format(CultureInfo.InvariantCulture, "multipart/mixed;boundary=batchresponse_{0}", Guid.NewGuid());
+
+            // Set the Content-Type header for existing batch formats
+            if (contentType == null)
+            {
+                contentType = String.Format(
+                    CultureInfo.InvariantCulture, "multipart/mixed;boundary=batchresponse_{0}", Guid.NewGuid());
```

Comment:
It seems this line code is repeating the line 200 in ODataBatchHttpRequestMessage.


Suggest to:

1. Add a helper method to create the default contentType
2. Use default in ODataBatchHttpRequestMessage "else"
3. line 34, use "default"
4. in constructor line 45, check "contentType"

Reply:
repeated.

---
In reply to: [202868720](https://github.com/OData/WebApi/pull/1536#discussion_r202868720) [](ancestors = 202868720)